### PR TITLE
Add System.IO.Compression to ExternalVersions

### PR DESCRIPTION
**Pull request by chcosta**
Currently, <s>System.Net.Http, System.Net.Security</s>, and System.IO.Compression are not going to ship from CoreFx, add them to the ExternalVersions regex list so that we don't try to reference them as CoreFx packages in our test builds (against packages).

/cc @jhendrixMSFT @weshaggard

### CODE REVIEW

**Comment by stephentoub** [comment]
> are not going to ship from CoreFx

What does this mean?

File: dir.props [hunk 1]
```diff
@@ -91,7 +91,7 @@
       <ExpectedPrerelease>$(CoreFxExpectedPrerelease)</ExpectedPrerelease>
     </ValidationPattern>
     <ValidationPattern Include="ExternalVersions">
-      <IdentityRegex>^(?i)System\.Data\.SqlClient$</IdentityRegex>
+      <IdentityRegex>^(?i)((System\.Data\.SqlClient)|(System\.IO\.Compression)|(System\.Net\.Http)|(System\.Net\.Security))$</IdentityRegex>
```
Comment (weshaggard) [inline dir.props]: As I sent en an offline email I would like to limit this to only partial Compression but for the other 2 I don't think we should do exclude these from the open builds.

**Comment by chcosta** [comment]
There are some packages which we haven't open sourced (fully), and won't ship from the sources on GitHub (though they may consume those sources).  ClrCompression is a good example where the sources for the packages we publish are still internal.  The others I mentioned, are still under discussion and may or may not fit into this category but got lumped in the same category because they existed in a data cell on a spreadsheet somewhere.

**Comment by chcosta** [comment]
@dotnet-bot test Innerloop CentOS7.1 Debug Build and Test

**Comment by jhendrixMSFT** [comment]
LGTM

**Pull request merged**